eo-maven-plugin/src/main/java/org/eolang/maven/FtCached.java

```diff
@@ -68,19 +68,19 @@ public final class FtCached implements Footprint {
 
     /**
      * Ctor.
-     * @param hash Version tag
+     * @param labels Hash and version of maven plugin
      * @param main Main root
      * @param cache Cache root
      */
-    public FtCached(final String hash, final Path main, final Path cache) {
-        this.hash = hash;
+    public FtCached(final String[] labels, final Path main, final Path cache) {
+        this.code = String.join("/", labels);
         this.main = main;
         this.cache = cache;
     }
 
     @Override
     public String load(final String program, final String ext) throws IOException {
-        final Path cached = new Place(program).make(this.cache.resolve(this.hash), ext);
+        final Path cached = new Place(program).make(this.cache.resolve(this.code), ext);
```

Comment:
@yasamprom here we want resolve the path from labels. Maybe new private method to be created and reused.